main.cpp

```diff
@@ -0,0 +1,59 @@
+#include <iostream>
+#include <random>
+#include <regex>
+
+void roll(int numberDice, int diceSides);
+void parseString(char *argument);
+
+int main(int argc, char *argv[]){
+//TODO Add some basic error checking
+    if(argc <= 1){
+        std::cout << "You haven't entered any arguments" << std::endl;
+        return -1;
+    }
+
+    for(int index = 1; index < argc; index++)
+         parseString(argv[index]);
+    return 0;
+}
+
+//Parses input then calls roll() on the arguments
+void parseString(char *argument){
+    //TODO Need to refactor, so unit testing is a viable strategy, possibly into class?
+    std::cmatch searchResults;
+
+    //Match any numbers at the begging of the line
+    std::regex firstNumberSearchString("^\\d+");
+
+    //Match any numbers at the end of line
+    std::regex secondNumberSearchString("\\d+$");
+
+    //Perform first search
+    regex_search(argument, searchResults, firstNumberSearchString);
```

Comment:
You should be checking the return value here to make sure it actually found a match before trying to convert it to an integer.